vendmachine/VendingMachine.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.BufferedWriter;
+import java.io.BufferedReader;
+import java.io.IOException;
+
+class VendingMachine {
+    private HashMap<String, Product> products = new HashMap<>();
+    private BufferedWriter log = null;
+    private String user = null;
+
+    private void close() {
+        try {
+            BufferedWriter state = new BufferedWriter(new FileWriter("state.csv"));
+            Iterator it = products.values().iterator();
+            while (it.hasNext()) {
+                Product p = (Product) it.next();
+                state.write(p.getName()+", "+p.getPrice()+", "+p.getDescription()+"\n");
+            }
+            state.close();
```

Comment:
🐛  There may be an issue here while an exception is thrown, the state will not close. My recommendation would be to move it to `finally`  block.

💡 There is also an easier way to `try-with-resources` statement as explained in details here: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html